### DISCUSSION TITLE
Can a program print without a newline?

### DISCUSSION THREAD

**Discussion by mortax** [grain-lang/grain#1602] [Help] [answered]
After reviewing the documentation and searching the internet, I have not been able to find any examples of printing a value without a newline.

Is that possible?

**Comment by spotandjake** [comment, marked as answer]
#666 Here is a reference to an issue covering this and how todo it, it sounds like the eventual hope will be a `sprintf` library that offers things like this. That issue covers a current way of actually printing without new line though.

**Reply by phated** [reply]
@spotandjake is correct. Grain is a WASI-compatible language, so you can just write to the stdout file descriptor.

**Reply by mortax** [reply]
i was just rereading the documentation and have a stylistic question about the solution provided in #666. which is considered better, the original suggestion:

```
File.fdWrite(File.stdout, stringvalue)
void
```

or this alternative:

```
ignore(File.fdWrite(File.stdout, stringvalue))
```

thanks!

**Reply by av8ta** [reply]
@mortax My understanding is if you're not concerned about an error when writing then using ignore is fine. Like if you're just printing to stdout for debugging. You might want to change `ignore` to `match` for handling errors in a "proper" app.

**Reply by ospencer** [reply]
Both are fine, but similar to what @av8ta is saying here, the most idiomatic version would be having a pattern match or using `Result.expect`:

```
let printString = (s) => {
  match (File.fdWrite(File.stdout, s)) {
    Ok(_) => void,
    Err(e) => ...
  }
}
```

```
let printString = (s) => {
  ignore(Result.expect("failed to print", File.fdWrite(File.stdout, s)))
}
```

If you're just asking about `void` versus `ignore`, that's mostly personal preference 🙂

**Reply by mortax** [reply]
that makes good sense. thanks.

**Reply by spotandjake** [reply]
Following up on this, As of `v0.6.0` we now support `print("Hello", suffix="")` which will omit the newline.
